app/models/registration.rb

```diff
@@ -445,4 +455,44 @@ def ensure_waitlist_eligibility!
   def serializable_hash(options = nil)
     super(DEFAULT_SERIALIZE_OPTIONS.merge(options || {}))
   end
+
+  def auto_accept
+    return log_error('Auto-accept is not enabled for this competition.') unless competition.auto_accept_registrations
```

Comment:
```suggestion
    return log_error('Auto-accept is not enabled for this competition.') unless competition.auto_accept_registrations?
```

Reply:
On a databse level, booleans are actually just `TINYINT(1)`, i.e. boolean-esque values encoded as either `0` or `1`. Using `if competition.auto_accept_registrations` relies on the fact that `0` "accidentally" behaves like `false` and `1` "accidentally" behaves like `true`.
With the question mark, Rails will properly convert it for you.

Reply:
Updated